internal/controller/gitrepository_controller.go

```diff
@@ -718,24 +689,38 @@ func (r *GitRepositoryReconciler) getAuthOpts(ctx context.Context, obj *sourcev1
 			conditions.MarkTrue(obj, sourcev1.FetchFailedCondition, e.Reason, "%s", e)
 			return nil, e
 		}
-
+		targetURL := fmt.Sprintf("%s://%s", u.Scheme, u.Host)
+		authMethods, err := secrets.AuthMethodsFromSecret(ctx, secret, secrets.WithTargetURL(targetURL), secrets.WithTLSSystemCertPool())
+		if err != nil {
+			return nil, err
+		}
+		if !authMethods.HasGitHubAppData() {
+			e := serror.NewStalling(
```

Comment:
While reviewing this specific line, I noticed that we are stalling in a few places in this controller where we should not, so I fixed it here:

https://github.com/fluxcd/source-controller/pull/1865

This is also a place where we should not stall. The secret can be updated and the controller will only know if it does not stall.

Stalling means adding the `Stalled` condition in the CR status and not enqueueing any more reconciliation requests. When this is done, the same CR is only reconciled again if the watcher receives an event. But if the secret with the GitHub App info is updated and fixed, we will not get any watch events, as the controller is not watching secrets.